File: patch_index.py
with open('api/src/index.ts', 'r') as f:
    content2 = f.read()

content2 = content2.replace("import { sendDailySummary, sendHourlySummary } from './telegram';", "import { sendDailySummary, sendHourlySummary, getDailySummaryText } from './telegram';")

new_endpoint = """
app.get('/api/bot/daily-text', async (c) => {
  const text = await getDailySummaryText();
  return c.json({ text });
});

"""

content2 = content2.replace("app.get('/api/cron/telegram/hourly'", new_endpoint + "app.get('/api/cron/telegram/hourly'")

with open('api/src/index.ts', 'w') as f:
    f.write(content2)

File: patch-index.js
const fs = require('fs');
let code = fs.readFileSync('api/src/index.ts', 'utf8');
code = code.replace(
  "import { listCollection, createDocument, updateDocument, deleteDocument } from './project-admin-proxy';",
  "import { listCollection, createDocument, updateDocument, deleteDocument, getDocument } from './project-admin-proxy';"
);

const newRoute = `// GET a single document
app.get('/api/project-admin/:projectId/:collection/:docId', async (c) => {
  try {
    const projectId = c.req.param('projectId');
    const collection = c.req.param('collection');
    const docId = c.req.param('docId');
    const databaseId = c.req.query('databaseId') || undefined;
    const doc = await getDocument(projectId, collection, docId, databaseId);
    if (!doc) return c.json({ error: 'Document not found' }, 404);
    return c.json({ data: doc });
  } catch (error: any) {
    console.error(\`Project Admin GET doc error:\`, error);
    return c.json({ error: error.message }, 500);
  }
});

app.post('/api/project-admin/:projectId/:collection',`;

code = code.replace("app.post('/api/project-admin/:projectId/:collection',", newRoute);

fs.writeFileSync('api/src/index.ts', code);
